src/main/java/org/opentripplanner/middleware/triptracker/TripInstruction.java

```diff
@@ -1,30 +1,217 @@
+            // High confidence.
+            double distanceToNextStep = Math.round(getDistance(activeStep.end, activeTripSegment.end));
+            return getInstruction(
+                distanceToNextStep,
+                getStepInstruction(expectedLeg.steps, activeStep.stepIndex + 1)
+            );
+        }
+
+        StepSegment nearestStep = getNearestStep(expectedLeg, activeTripSegment);
+
+        if (nearestStep != null) {
+            // Lower confidence. This part could benefit from "traveler bearing processing".
+            Coordinates destination = new Coordinates(expectedLeg.to);
+            if (isLastStep(nearestStep.stepIndex, expectedLeg)) {
+                if (!isTripSegmentNearerToDestination(nearestStep, activeTripSegment, destination)) {
+                    // Assuming traveler is approaching the last step.
+                    return approachingStep(activeTripSegment, nearestStep, expectedLeg);
+                }
+            } else {
+                if (isTripSegmentNearerToDestination(nearestStep, activeTripSegment, destination)) {
+                    // Assuming traveler is passed the nearest step.
+                    Step nextStep = expectedLeg.steps.get(nearestStep.stepIndex + 1);
+                    return getInstruction(
+                        getDistance(activeTripSegment.end, new Coordinates(nextStep)),
+                        nextStep.relativeDirection
+                    );
+                } else {
+                    // Assuming traveler is approaching the nearest step.
+                    return approachingStep(activeTripSegment, nearestStep, expectedLeg);
+                }
+            }
+        }
+        return TripInstruction.NO_INSTRUCTION;
+    }
+
+    /**
+     * Does the step index reference the last step in the leg.
+     */
+    private static boolean isLastStep(int stepIndex, Leg expectedLeg) {
+        return stepIndex == (expectedLeg.steps.size() - 1);
+    }
+
+    /**
+     * Provide instruction for approaching a step.
+     */
+    private static String approachingStep(TripSegment activeTripSegment, StepSegment nearestStep, Leg expectedLeg) {
+        return getInstruction(
+            getDistance(activeTripSegment.end, nearestStep.start),
+            getStepInstruction(expectedLeg.steps, nearestStep.stepIndex)
+        );
+    }
+
+    /**
+     * Determine if the trip segment is near to the destination than the step.
+     */
+    private static boolean isTripSegmentNearerToDestination(
+        StepSegment stepSegment,
+        TripSegment tripSegment,
+        Coordinates destination
+    ) {
+        double distanceFromLastStepToDestination = getDistance(stepSegment.start, destination);
+        double distanceFromActiveSegmentToDestination = getDistance(tripSegment.start, destination);
+        return distanceFromActiveSegmentToDestination < distanceFromLastStepToDestination;
+    }
+
+    /**
+     * Get instruction based distance and related step instruction.
```

Comment:
Can you add an example of how that should look, please. I would assume this is a formatted code as opposed to human-readable content?